### ISSUE TITLE
[Bug]: devbox keeps trying to install mysql80 but fails when the services are already running

### ISSUE THREAD

**Issue by calvinbaart** [jetpack-io/devbox#1612]
**Current Behavior (bug)**
When we start the services using devbox services up and then in another shell do devbox shell we get this error:

```
➜  test2 devbox shell

Installing package: path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80.

[1/1] path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80
Error: error installing package path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80: error searching for pkg path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80: exit status 1

Error: There was an internal error. Run with DEVBOX_DEBUG=1 for a detailed error message, and consider reporting it at https://github.com/jetpack-io/devbox/issues
```

When running with DEVBOX_DEBUG=1 we get this log:

```
➜  test2 DEVBOX_DEBUG=1 devbox shell
2023/11/08 09:50:29 findProjectDir: path is
2023/11/08 09:50:29 finding devbox.json in dir: /Users/calvinbaart/dev/test2
2023/11/08 09:50:29 findProjectDir: path is
2023/11/08 09:50:29 finding devbox.json in dir: /Users/calvinbaart/dev/test2
2023/11/08 09:50:29 findProjectDir: path is
2023/11/08 09:50:29 finding devbox.json in dir: /Users/calvinbaart/dev/test2
2023/11/08 09:50:29 Creating files for package "mysql80@8.0.34" create files
2023/11/08 09:50:29 Creating file "/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80/setup_db.sh" from contentPath: "mysql/setup_db.sh"
2023/11/08 09:50:29 Creating file "/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80/process-compose.yaml" from contentPath: "mysql/process-compose.yaml"
2023/11/08 09:50:29 Creating file "/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80/flake.nix" from contentPath: "mysql/flake.nix"

Installing package: path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80.

[1/1] path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80
2023/11/08 09:50:29 running command: /nix/var/nix/profiles/default/bin/nix search --json path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80 --extra-experimental-features ca-derivations --option experimental-features nix-command flakes fetch-closure
Error: error installing package path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80: error searching for pkg path:/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80: exit status 1

2023/11/08 09:50:30 Command stderr: error (ignored): error: reached end of FramedSource
error: file '/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80/run/mysql.sock' has an unsupported type
(use '--show-trace' to show detailed location information)

2023/11/08 09:50:30
ExecutionID:d7a3227c0579426da751f81cdffcba10
<nil>
2023/11/08 09:50:30 findProjectDir: path is
2023/11/08 09:50:30 finding devbox.json in dir: /Users/calvinbaart/dev/test2
```

We get the same error afterwards when closing the services manager. This seems to happen because the process is not exited correctly:

```
➜  test2 ps -A | grep mysql
34225 ttys003    0:00.09 nix search --json path:/Users/calvinbaart/dev/backend/.devbox/virtenv/mysql80 --extra-experimental-features ca-derivations --option experimental-features nix-command flakes fetch-closure
35533 ttys003    0:01.06 /Users/calvinbaart/dev/test2/.devbox/nix/profile/default/bin/mysqld --datadir=/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80/data --pid-file=/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80/run/mysql.pid --socket=/Users/calvinbaart/dev/test2/.devbox/virtenv/mysql80/run/mysql.sock
35884 ttys003    0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox mysql
```

**Expected Behavior (fix)**
We expect it to just open the shell without issues. This happens on every command we execute (devbox run ..., devbox add, devbox shell etc.). Before devbox 0.8.1 we could workaround this issue by just pressing CTRL+C when `Ensuring packages are installed` showed up. When we run the command after that it seems to skip that step. This doesn't happen anymore in 0.8.1 so we have no workaround.

**Additional context**
```
➜  test2 devbox version -v
Version:     0.8.1
Platform:    darwin_arm64
Commit:      a0bdff918b3dde5ebcfc488f72df5efdf3426253
Commit Time: 2023-11-07T21:43:27Z
Go Version:  go1.21.3
Launcher:    0.2.0
```

devbox.json:
```json
{
  "packages": [
    "mysql80@8.0.34"
  ],
  "shell": {
    "init_hook": [
      "echo 'Welcome to devbox!' > /dev/null"
    ],
    "scripts": {
      "test": [
        "echo \"Error: no test specified\" && exit 1"
      ]
    }
  }
}
```

**Comment by skyosev** [comment]
Same here. There seems to be an issue with mysql (or more general one).

**Comment by jrseliga** [comment]
@mikeland73 is this resolved by https://github.com/jetpack-io/devbox/pull/1625?

**Comment by Lagoja** [comment]
@jrseliga I think this and #1626 should address the issue.

**Comment by Lagoja** [comment]
This should be resolved in #1626, which was released in 0.8.2. Closing, but will reopen if the issue reappears
